src/Draco.Compiler/Internal/Syntax/Parser.cs

```diff
@@ -261,6 +261,13 @@ private DeclarationSyntax ParseDeclaration(DeclarationContext context)
         switch (this.Peek())
         {
         case TokenKind.KeywordImport:
+            if (modifier is not null)
+            {
+                var info = DiagnosticInfo.Create(SyntaxErrors.UnexpectedVisibilityModifierBeforeImport, formatArgs: "declaration");
+                var diag = new SyntaxDiagnosticInfo(info, Offset: 0, Width: modifier.Width);
+                this.AddDiagnostic(modifier, diag);
+                return new UnexpectedDeclarationSyntax(modifier, SyntaxList.Create(Array.Empty<SyntaxNode>()));
+            }
             return this.ParseImportDeclaration();
```

Comment:
A low hanging fruit would be to store the invalid token right in the import statement so the shape would be
```
ImportDeclaration
{
    IllegalToken(public),
    rest of the node...
}
```
Better tree shape overall.

Reply:
The purpose was that the import would still be entirely valid and be correctly parsed.